File: backend/src/main/java/hu/bsstudio/raktr/dto/rent/RentUpdateDto.java
package hu.bsstudio.raktr.dto.rent;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RentUpdateDto {

    @NotBlank
    private String destination;

    @NotNull
    private UUID issuerId;

    @NotBlank
    private String renterName;

    @NotNull
    private LocalDate outDate;

    @NotNull
    private LocalDate expectedReturnDate;

    private LocalDate actualReturnDate;

}

File: frontend/src/app/model/rent/rentUpdateDto.ts
export class RentUpdateDto {
  destination: string;
  issuerId: string;
  renterName: string;
  outDate: Date;
  expectedReturnDate: Date;
  actualReturnDate: Date | null;

  constructor(
    destination: string,
    issuerId: string,
    renterName: string,
    outDate: Date,
    expectedReturnDate: Date,
    actualReturnDate: Date | null,
  ) {
    this.destination = destination;
    this.issuerId = issuerId;
    this.renterName = renterName;
    this.outDate = outDate;
    this.expectedReturnDate = expectedReturnDate;
    this.actualReturnDate = actualReturnDate;
  }

  static fromJson(json: Record<string, unknown>): RentUpdateDto {
    return new RentUpdateDto(
      json['destination'] as string,
      json['issuerId'] as string,
      json['renterName'] as string,
      new Date(json['outDate'] as string),
      new Date(json['expectedReturnDate'] as string),
      json['actualReturnDate'] ? new Date(json['actualReturnDate'] as string) : null,
    );
  }

  toJson(): Record<string, unknown> {
    return {
      destination: this.destination,
      issuerId: this.issuerId,
      renterName: this.renterName,
      outDate: this.outDate.toISOString().split('T')[0],
      expectedReturnDate: this.expectedReturnDate.toISOString().split('T')[0],
      actualReturnDate: this.actualReturnDate
        ? this.actualReturnDate.toISOString().split('T')[0]
        : null,
    };
  }
}
